lib/python_helper_sympy.py

```diff
@@ -210,7 +283,19 @@ def evaluate(expr, locals_for_eval={}):
     # https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html
     # http://blog.delroth.net/2013/03/escaping-a-python-sandbox-ndh-2013-quals-writeup/
     #
-    whitelist = (ast.Module, ast.Expr, ast.Load, ast.Expression, ast.Call, ast.Name, ast.Num, ast.UnaryOp, ast.UAdd, ast.USub, ast.BinOp, ast.Add, ast.Sub, ast.Mult, ast.Div, ast.Mod, ast.Pow)
+    # FOR LATER:
+    # _safe_nodes = [
+    #    'Add', 'And', 'BinOp', 'BitAnd', 'BitOr', 'BitXor', 'BoolOp',
+    #    'Compare', 'Dict', 'Eq', 'Expr', 'Expression', 'For',
+    #    'Gt', 'GtE', 'Is', 'In', 'IsNot', 'LShift', 'List',
+    #    'Load', 'Lt', 'LtE', 'Mod', 'Name', 'Not', 'NotEq', 'NotIn',
+    #    'Num', 'Or', 'RShift', 'Set', 'Slice', 'Str', 'Sub',
+    #    'Tuple', 'UAdd', 'USub', 'UnaryOp', 'boolop', 'cmpop',
+    #    'expr', 'expr_context', 'operator', 'slice', 'unaryop' 
+    # ]
```

Comment:
Are you sure these are all safe to evaluate (later, if implemented)? I know there are all kinds of Python sandbox escapes. In particular, I wonder about `Dict` and the expression-related ones (`Expr`, `expr`, etc.).

Reply:
@echuber2 
I found that list digging around stack exchange while I was working on this. Those should hit the standard Python operations. The problem that I ran into was that not all types (e.g., ``scipy``) get converted into something easy to interface to ``sympy`` with by default. 

What I was actually thinking with that list is to implement some custom relations. This is half a joke, and only partially working code right now, but the "*ternary snowflake operator*" (of discrete math fame) is an idea. The definition is taken to be that "a ❄ b ❆ c" returns the following (the operators are unicode and display better inline with MacVim):
 * -> 0 if a === 0 (mod 2)
 * -> 1 if b === 0 (mod 2)
 * -> c otherwise

[TernarySnowflake.py.txt](https://github.com/PrairieLearn/PrairieLearn/files/7230638/TernarySnowflake.py.txt)

I think it's a cool idea for later that may be as useful as the custom function names for some applications. You can delete the comment I added about standard Python AST items to handle.

Reply:
Moving the param to the end is fine with me if you want to make that modification.

Reply:
I think the issue I wanted to reiterate was just that we really don't want to support a complete Python evaluation in the Sympy handler, for security and stability. In more advanced use cases, courses may be better off making a Workspace container.

Reply:
> Moving the param to the end is fine with me if you want to make that modification.

This is from the other review convo, I think. There's the bigger question of whether to merge the two functions in the refactor.